frontend/src/test/scala/bloop/bsp/BspMetalsClientSpec.scala

```diff
@@ -0,0 +1,399 @@
+package bloop.bsp
+
+import bloop.io.AbsolutePath
+import bloop.cli.BspProtocol
+import bloop.util.TestUtil
+import bloop.util.TestProject
+import bloop.logging.RecordingLogger
+import bloop.logging.BspClientLogger
+import bloop.cli.ExitStatus
+import bloop.data.WorkspaceSettings
+import bloop.internal.build.BuildInfo
+import bloop.bsp.BloopBspDefinitions.BloopExtraBuildParams
+
+import java.nio.file.Files
+
+import io.circe.JsonObject
+import io.circe.Json
+
+import monix.execution.Scheduler
+import monix.execution.ExecutionModel
+import monix.eval.Task
+
+import scala.concurrent.duration.FiniteDuration
+
+import ch.epfl.scala.bsp.endpoints.BuildTarget.scalacOptions
+
+object LocalBspMetalsClientSpec extends BspMetalsClientSpec(BspProtocol.Local)
+object TcpBspMetalsClientSpec extends BspMetalsClientSpec(BspProtocol.Tcp)
+
+class BspMetalsClientSpec(
+    override val protocol: BspProtocol
+) extends BspBaseSuite {
+  private val testedScalaVersion = "2.12.8"
+
+  test("initialize metals client and save settings") {
+    TestUtil.withinWorkspace { workspace =>
+      val `A` = TestProject(workspace, "A", Nil, scalaVersion = Some(testedScalaVersion))
+      val projects = List(`A`)
+      val configDir = TestProject.populateWorkspace(workspace, projects)
+      val logger = new RecordingLogger(ansiCodesSupported = false)
+      val semanticdbVersion = "4.2.0"
+      val extraParams = BloopExtraBuildParams(
+        clientClassesRootDir = None,
+        semanticdbVersion = Some(semanticdbVersion),
+        supportedScalaVersions = List(testedScalaVersion)
+      )
+
+      loadBspState(workspace, projects, logger, "Metals", bloopExtraParams = extraParams) { state =>
+        assertNoDiff(logger.warnings.mkString(System.lineSeparator), "")
+        assertNoDiffInSettingsFile(
```

Comment:
In general, I really like testing against strings and showing diffs to know what's wrong opposed to doing concrete assertions. This way, it's obvious when reading the tests what the output of workspace settings should be. Just a FYI :smile: